framework/framework.go

```diff
@@ -42,6 +44,7 @@ type Config struct {
 	// and different resources can be executed depending on the runtime object
 	// being reconciled.
 	ResourceRouter *ResourceRouter
+	RestClient     rest.Interface
```

Comment:
This needs to be `rest.Interface` otherwise the fake client can't be used. Unsure if RestClient is the correct naming here.

Reply:
It wouldn't be bad if it was `kubernetes.Interface`. Maybe we'll be adding some functionality in the future so the interface won't have to change.

Reply:
I don't like using `kubernetes.Interface`, `rest.Interface` does exactly what we need here. Passing the `kubernetes.Interface` just to then get the `rest.Interface` out of it doesn't feel clean to me.

WDYT?

Reply:
This is forward thinking. Lets get the k8s interface in and then we have it without the need of interface changes later on again when we suddenly realize we really need it now. Then we are already safe. Thinking about the future this makes a lot of sense to me.

Reply:
Done.

I assume that the `RESTClient` is never `nil` if the `K8sClient` is not nil. Or is this a bad assumption?

Reply:
I think it is safe enough to assume there is a rest client in case there is a k8s client. When this is not true it is badly mocked or something went horribly wrong in the upstream code generation.